src/utils/body.ts

```diff
@@ -18,20 +19,37 @@ export type ParseBodyOptions = {
    * parseBody should return { file: ['aaa', 'bbb'], message: 'hello' }
    */
   all?: boolean
+  /**
+   * Determines whether all fields with dot notation should be parsed as nested objects.
+   * @default false
+   * @example
+   * const data = new FormData()
+   * data.append('obj.key1', 'value1')
+   * data.append('obj.key2', 'value2')
+   *
+   * If dot is false:
+   * parseBody should return { 'obj.key1': 'value1', 'obj.key2': 'value2' }
+   *
+   * If dot is true:
+   * parseBody should return { obj: { key1: 'value1', key2: 'value2' } }
+   */
+  dot?: boolean
 }
 
 export const parseBody = async <T extends BodyData = BodyData>(
   request: HonoRequest | Request,
-  options: ParseBodyOptions = { all: false }
+  options: ParseBodyOptions = {}
 ): Promise<T> => {
+  const { all = options?.all ?? false, dot = options?.dot ?? false } = options
+
   const headers = request instanceof HonoRequest ? request.raw.headers : request.headers
   const contentType = headers.get('Content-Type')
 
   if (
     (contentType !== null && contentType.startsWith('multipart/form-data')) ||
     (contentType !== null && contentType.startsWith('application/x-www-form-urlencoded'))
   ) {
-    return parseFormData<T>(request, options)
+    return parseFormData<T>(request, { ...options, all, dot })
```

Comment:
Then this will be `{ all, dot }`

Reply:
Sure

Reply:
```js
if (
    contentType !== null && (contentType.startsWith('multipart/form-data')) || contentType.startsWith('application/x-www-form-urlencoded')
  )
```

Reply:
Is something wrong with the contentType? @vanodevium 

I believe we need to wrap it like this, right?

Since OR have two conditions, so we need to use parentheses

```ts
if (
  contentType !== null && ((contentType.startsWith('multipart/form-data')) || contentType.startsWith('application/x-www-form-urlencoded'))
)
```

Reply:
```js
if (
        contentType !== null && (contentType.startsWith('multipart/form-data') || contentType.startsWith('application/x-www-form-urlencoded'))
    )
```